### PR TITLE
smbutil.c: Replace obsolete function call (asctime)

### PR DESCRIPTION
**Pull request by egecetin**

### CODE REVIEW

File: smbutil.c [hunk 1]
```diff
@@ -800,10 +801,11 @@ smb_fdata1(netdissect_options *ndo,
 	    }
 	    if (t != 0) {
 		lt = localtime(&t);
-		if (lt != NULL)
-		    tstring = asctime(lt);
-		else
-		    tstring = "(Can't convert time)\n";
+		if (lt != NULL) {
+	        strftime(buffer, 26, "%a %b %e %T %Y%n", lt);;
```
Comment (infrastation) [inline smbutil.c:805]: * `sizeof(buffer)`
* without the extra semicolon please

File: smbutil.c [hunk 2]
```diff
@@ -800,10 +801,11 @@ smb_fdata1(netdissect_options *ndo,
...
+	        tstring = buffer;
+	    } else
+		tstring = "(Can't convert time)\n";
```
Comment (infrastation) [inline smbutil.c:808]: Did you mean to change this line?

**Comment by infrastation** [comment]
See also the-tcpdump-group/tcpslice@e3d63a6.

File: smbutil.c (hunk 2 above)
Comment (egecetin) [reply smbutil.c:808]: TBH not understand what you mean, sorry

Comment (guyharris) [reply smbutil.c:808]: > > Did you mean to change this line?
> TBH not understand what you mean, sorry

The code before your change is:

 		lt = localtime(&t);
 		if (lt != NULL)
 		    tstring = asctime(lt);
 		else
 		    tstring = "(Can't convert time)\n";

The code after your change is:

 		lt = localtime(&t);
 		if (lt != NULL) {
 	        strftime(buffer, sizeof(buffer), "%a %b %e %T %Y%n", lt);
 	        tstring = buffer;
 	    } else
 		tstring = "(Can't convert time)\n";

The code using `strftime()` *should* be:

 		lt = localtime(&t);
 		if (lt != NULL) {
 	            strftime(buffer, sizeof(buffer), "%a %b %e %T %Y%n", lt);
 	            tstring = buffer;
 		} else
 		    tstring = "(Can't convert time)\n";

which is correctly indented, and doesn't change the lines after the `asctime()` call.

Comment (egecetin) [reply smbutil.c]: Understand now thanks!

File: smbutil.c [hunk 3]
```diff
@@ -770,6 +770,7 @@ smb_fdata1(netdissect_options *ndo,
 	    time_t t;
 	    struct tm *lt;
 	    const char *tstring;
+	    char buffer[26];
```
Comment (fxlb) [inline smbutil.c:773]: "26" must be documented.

Comment (egecetin) [reply smbutil.c]: Done!

Comment (fxlb) [reply smbutil.c]: Using @infrastation hint, I think it's better to use:
`char buffer[sizeof("Www Mmm dd hh:mm:ss yyyy\n")];`

**Comment by fxlb** [comment]
Using @infrastation hint, I think it's better to use:
`char buffer[sizeof("Www Mmm dd hh:mm:ss yyyy\n")];`

**Comment by infrastation** [comment]
Whichever way this eventually settles, please remember to squash the commits together before merging.

**Comment by egecetin** [comment]
Ready to merge if ok for everyone

**Comment by infrastation** [comment]
Please do not forget to update `CHANGES` and `CREDITS`.

**Comment by egecetin** [comment]
Thanks for the notice. I missed it because this is my first contribution to this repository.

**Comment by fxlb** [comment]
Please undo the merge, do a rebase.

File: CHANGES [hunk 4]
```diff
@@ -51,6 +51,8 @@ TBD
       Cirrus CI: Run the "make releasecheck" command in the Linux task.
       Makefile.in: Add the whitespacecheck target.
       Cirrus CI: Run the "make whitespacecheck" command in the Linux task.
+    Source code:
+      smbutil.c: Replace obsolote function call (asctime)
```
Comment (fxlb) [inline CHANGES:55]: Typo here: obsolote

**Comment by fxlb** [comment]
It's better to amend the commit message to have `smbutil.c: Replace obsolete function call (asctime)` like in CHANGES.

**Comment by fxlb** [comment]
Thank you!

**Pull request merged**